icu4j/main/classes/core/src/com/ibm/icu/message2/Mf2DataModelFormatter.java

```diff
@@ -0,0 +1,277 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.ibm.icu.message2.Mf2DataModel.Expression;
+import com.ibm.icu.message2.Mf2DataModel.Part;
+import com.ibm.icu.message2.Mf2DataModel.Pattern;
+import com.ibm.icu.message2.Mf2DataModel.SelectorKeys;
+import com.ibm.icu.message2.Mf2DataModel.Text;
+import com.ibm.icu.message2.Mf2DataModel.Value;
+import com.ibm.icu.text.FormattedValue;
+import com.ibm.icu.util.Calendar;
+import com.ibm.icu.util.CurrencyAmount;
+
+/**
+ * Takes an {@link Mf2DataModel} and formats it to a {@link String}
+ * (and later on we will also implement formatting to a {@code FormattedMessage}).
+ */
+// TODO: move this in the MessageFormatter
+class Mf2DataModelFormatter {
+    private final Locale locale;
+    private final Mf2DataModel dm;
+
+    final Mf2FunctionRegistry standardFunctionRegistry;
+    final Mf2FunctionRegistry customFunctionRegistry;
```

Comment:
totally optional: s/...FunctionRegistry/...Functions/ to make the code a little shorter, which I think will make it a little more readable

Reply:
Done